vividus-plugin-web-app/src/main/java/org/vividus/steps/ui/web/ElementCssSteps.java

```diff
@@ -0,0 +1,237 @@
+            replacementFormatPattern = "Then context element has CSS property `%1$s` with value that contains `%2$s`")
+    @Then("the context element has the CSS property '$cssName' containing '$cssValue'")
+    public void doesElementHaveRightPartOfCssValue(String cssName, String cssValue)
+    {
+        uiContext.getSearchContext(WebElement.class).ifPresent(element -> {
+            String actualCssValue = webElementActions.getCssValue(element, cssName);
+
+            descriptiveSoftAssert.assertThat("Css property value part is correct",
+                    String.format(ELEMENT_CSS_CONTAINING_VALUE, cssName, cssValue),
+                    actualCssValue, containsString(cssValue));
+        });
+    }
+
+    /**
+     * Checks that the context <b>element</b> has an expected <b>CSS properties</b>
+     * <p>The expected CSS parameters to be defined in the ExamplesTable:</p>
+     * <ul>
+     * <li><b>cssName</b> - the name of the CSS property</li>
+     * <li><b>comparisonRule</b> - String comparison rule: "is equal to", "contains", "does not contain",
+     * "matches".</li>
+     * <li><b>expectedValue</b> - expected CSS property value</li>
+     * </ul>
+     * <p>Usage example:</p>
+     * <code>
+     * <br>Then context element has CSS properties:
+     * <br>|cssName |comparisonRule |expectedValue |
+     * <br>|border  |contains       |solid         |
+     * </code>
+     *
+     * @param parameters The parameters of the expected CSS properties to set as ExamplesTable
+     */
+    @Then("context element has CSS properties:$parameters")
+    public void doesElementHasCssProperties(ExamplesTable parameters)
+    {
+        uiContext.getSearchContext(WebElement.class).ifPresent(element -> {
+            String getAllCssScript =
+                    org.vividus.util.ResourceUtils.loadResource("org/vividus/ui/web/get-element-computed-css-func.js");
+            getAllCssScript += "return getComputedStyleAsMap(arguments[0])";
+            Map<String, String> elementCss = javascriptActions.executeScript(getAllCssScript, element);
+
+            List<CssValidationResult> cssResults = validateElementCss(parameters, elementCss);
+            attachmentPublisher.publishAttachment("templates/css_validation_result.ftl",
+                    Map.of("cssResults", cssResults), "Css validation results");
+        });
+    }
+
+    private List<CssValidationResult> validateElementCss(ExamplesTable parameters, Map<String, String> elementCss)
+    {
+        List<Parameters> rowsAsParameters = parameters.getRowsAsParameters();
+        List<CssValidationResult> cssResults = new ArrayList<>();
+        rowsAsParameters.forEach(params ->
+        {
+            Map<String, String> values = params.values();
+            String cssName = values.get("cssName");
+            String expectedValue = values.get("expectedValue");
+            StringComparisonRule comparisonRule = params.valueAs("comparisonRule", StringComparisonRule.class);
```

Comment:
does it make sense to map params directly to pojo having css name, exp val and comp rule?

Reply:
fixed